### ISSUE TITLE
GREATER_THAN, LESS_THAN filter questions

### ISSUE THREAD

**Issue by Dev63** [angular-ui/ng-grid#2121]
Is uiGridConstants.filter.GREATER_THAN supposed to filter out null/empty values? For myself, I'd like to see them get filtered out, but can easily write my own filter if that's needed.

On a related note, how is one supposed to deal with negative numbers?  If I want to show only numbers less than -50, can I use the predefined filters somehow? When I type in a negative number, it doesn't seem to do anything.

**Comment by brockj** [comment]
The issue with negative numbers looks to be a different issue (and possibly not an enhancement)

It looks like the filter term  is being bound directly to the columns model in the header cell template as a string value using ng-model and then used in runColumnFilter to compare against the row values.

For the greater than and less than cases, would it be possible to check the value type to see if it is a number and if so attempt to convert the term to a number before comparing?  Ideally, the term would already be a number, but i suspect that is a much larger change.

As a quick test i added this to each of the filter conditions and it mostly worked (it compares against NaN when the input value is invalid or a minus sign).  It used filter.term because the term is escaped as a regex (minus becomes \-)
``` javascript
                if (typeof(value) === 'number' && typeof(filter.term) === 'string') {
                    term = +filter.term;
                }
```

**Comment by PaulL1** [comment]
I think I fixed this in an earlier PR, could you retest and confirm?

**Comment by Dev63** [comment]
@PaulL1 Yes, both items work great now (null is not > 5, and negative number filtering)!

**Comment by brockj** [comment]
I am not sure the negative number filtering is working.

given a column value of -20 i would like to fine numbers less than -10.  This currently doesn't seem to work.  Should i open a new issue for this?

**Comment by PaulL1** [comment]
Yeah, I'd suggest a new issue, and probably a plunker just to make it easier to work on.

**Comment by Dev63** [comment]
@PaulL1  and @brockj : Yes, I was too quick to confirm the negative number filtering worked.  Here is a [negative number filtering plunker](http://plnkr.co/edit/9Eq7XrxysJl9ILXP9kBa?p=preview) from the filtering demo, where I just changed a couple of ages to negative.

**Comment by PaulL1** [comment]
@Dev63: I think I fixed this this morning - could you re-verify.

**Comment by Dev63** [comment]
@PaulL1 OK, yes, now it really works.  It is a little disconcerting to have everything blank out when you first type the negative sign, though, and then reappear once you finish the number.  I'd suggest ignoring any '-' until you have something after it.

**Comment by c0bra** [comment]
Don't think we can do that. What if I'm searching for a minus sign?

**Comment by Dev63** [comment]
@c0bra Fortunately, we don't search for characters in a LT filter :-)   So if the LT filter gets just a '-', just return true for all elements.  Or I suppose you could deduce that only numbers < 0 will pass, and return them.

@PaulL1 One other thing to consider.  The GT filter now filters out nulls, but the LT filter does not, and treats a null as 0.  In my opinion, nulls should not match anything, but at a minimum, the filters should behave analogously.  

Sorry my initial testing was so lackluster; I guess I'm more geared towards writing code, not testing it!

**Comment by PaulL1** [comment]
I had thought they were both treating nulls the same - basically null seems to get treated as low_value, so it's less than everything and greater than nothing.  This is consistent with the way we sort, so I left it alone.

On the -, we do allow LT and GT on strings I think.  So until what you've typed becomes a number (which it isn't when it's only a - sign), it doesn't get filtered as a number.

**Comment by Dev63** [comment]
@PaulL1 The [negative number filtering plunker](http://plnkr.co/edit/9Eq7XrxysJl9ILXP9kBa?p=preview) I created shows that nulls never match a GT filter, but are treated as 0 for LT.  To see the nulls in the plunker, just unclick the gender filter.

Separately, I understand your point about '-' also being used for strings: I was myopically focused on my own use cases.  Sounds like if you wanted to solve the complaint I raised about disappearing and reappearing rows as one types a negative number filter, you'd want the filter to understand whether the column was a number or not (probably more trouble than it's worth), or provide a separate NLT (numeric less than) filter.  In any case, this is a very minor point, and I'd probably just let it drop.
